adoc/extensions/sycl_khr_free_function_commands.adoc

```diff
@@ -0,0 +1,928 @@
+
+}
+----
+_Effects (1)_: Equivalent to [code]#h.fill(ptr, pattern, count)#.
+
+_Effects (2)_: Equivalent to [code]#h.fill(dest, src)#.
+
+_Effects (3)_: Equivalent to [code]#q.submit([&](handler& h) { fill(h, ptr,
+pattern, count); })#.
+
+_Effects (4)_: Equivalent to [code]#q.submit([&](handler& h) { fill(h, dest,
+src); })#.
+
+'''
+
+.[apititle]#update_host#
+[source,role=synopsis,id=api:update_host]
+----
+namespace sycl::khr {
+
+template <typename T, int Dims, access_mode Mode>
+void update_host(sycl::handler& h, accessor<T, Dims, Mode, target::device> acc); (1)
+
+template <typename T, int Dims, access_mode Mode>
+void update_host(sycl::queue q, accessor<T, Dims, Mode, target::device> acc);    (2)
+
+}
+----
+_Constraints_: [code]#T# must be <<device-copyable>>.
+
+_Effects (1)_: Equivalent to [code]#h.update_host(acc)#.
+
+_Effects (2)_: Equivalent to [code]#q.submit([&](handler& h) { update_host(h,
+acc); })#.
+
+'''
+
+.[apititle]#prefetch#
+[source,role=synopsis,id=api:prefetch]
+----
+namespace sycl::khr {
+
+void prefetch(sycl::handler& h, void* ptr, size_t numBytes); (1)
+
+void prefetch(sycl::queue q, void* ptr, size_t numBytes);    (2)
+
+}
+----
+_Effects (1)_: Equivalent to [code]#h.prefetch(ptr, numBytes)#.
+
+_Effects (2)_: Equivalent to [code]#q.submit([&](handler& h) { prefetch(h, ptr,
+numBytes); })#.
+
+'''
+
+.[apititle]#mem_advise#
+[source,role=synopsis,id=api:mem_advise]
+----
+namespace sycl::khr {
+
+void mem_advise(sycl::handler& h, void* ptr, size_t numBytes, int advice); (1)
+
+void mem_advise(sycl::queue q, void* ptr, size_t numBytes, int advice);    (2)
+
+}
+----
+_Effects (1)_: Equivalent to [code]#h.mem_advise(ptr, numBytes, advice)#.
+
+_Effects (2)_: Equivalent to [code]#q.submit([&](handler& h) { mem_advise(h,
+ptr, numBytes, advice); })#.
+
+'''
+
+=== Command and event barriers
+
+.[apititle]#command_barrier#
+[source,role=synopsis,id=api:command_barrier]
+----
+namespace sycl::khr {
+
+void command_barrier(sycl::handler& h); (1)
+
+void command_barrier(sycl::queue q);    (2)
+
+}
+----
+_Effects_: Enqueues a command barrier.
+Any commands submitted after this barrier cannot begin execution until all
+previously submitted commands (and any commands associated with dependendent
+events) have completed.
+
+'''
+
+.[apititle]#event_barrier#
+[source,role=synopsis,id=api:event_barrier]
+----
+namespace sycl::khr {
+
+void event_barrier(sycl::handler& h, const std::vector<sycl::event>& events); (1)
+
+void event_barrier(sycl::queue q, const std::vector<sycl::event>& events);    (2)
+
+}
+----
+_Effects_: Enqueues an event barrier.
+Any commands submitted after this barrier cannot begin execution until all
+commands associated with [code]#events# (and any commands associated with other
+dependent events) have completed.
+
+{note}For both overloads, if [code]#events# is empty and an event barrier has no
+other dependencies (e.g., specified by [code]#handler::depends_on#), it is ot
```

Comment:
```suggestion
other dependencies (e.g., specified by [code]#handler::depends_on#), it is not
```

Reply:
Good catch!  Fixed in 9c627921aa959a199a29ce246491c5b995fdef58.